### PR TITLE
Edit reviews

### PR DESCRIPTION
**Pull request by gilhsu**
Created edit functionality to reviews all in react. Existing reviews edit in place on the show page and if the revision is valid, the edited review is persisted on the show page again. Also includes all relevant tests.

### CODE REVIEW

File: app/javascript/containers/ReviewContainer.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import React, { Component } from "react"
+import ReviewEditFormTile from "../tiles/ReviewEditFormTile";
+import ReviewTile from "../tiles/ReviewTile"
+
+
+
```
Comment (nwalberts) [inline app/javascript/containers/ReviewContainer.js]: remove extra spacing

File: app/javascript/containers/ReviewContainer.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      alert("there was a problem with the submission")
+    });
+  }
+
+  componentDidMount(){
+    this.setState({review: this.props.review})
+  }
+
+
+  render() {
```
Comment (nwalberts) [inline app/javascript/containers/ReviewContainer.js:67]: watch the extra spacing

File: app/javascript/containers/ReviewContainer.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    let review =()=>{
+      if(this.state.edit){
+        return <ReviewEditFormTile handleSubmit={this.handleSubmit} review={this.state.review} handleChange={this.handleChange}/>
+      } else{
+        return <ReviewTile review={this.state.review} onClick={this.handleEditClick} />
+      }
+    }
```
Comment (nwalberts) [inline app/javascript/containers/ReviewContainer.js:74]: this probably doesnt need to be a function. Just make it a `review` variable.

File: app/javascript/containers/ReviewsContainer.js [hunk 4]
```diff
@@ -9,6 +9,8 @@ class ReviewsContainer extends Component {
         };
     }
 
+
+
```
Comment (nwalberts) [inline app/javascript/containers/ReviewsContainer.js]: probably dont need this extra space

File: app/views/restaurants/index.html.erb [hunk 5]
```diff
@@ -5,8 +5,11 @@
     <li>
       <%=link_to(restaurant) do %>
         <%= restaurant.name %>
+        <% if current_user && current_user.admin %>
+          <%= link_to 'Delete', restaurant, :method => 'delete', class: "alert tiny button", data: { confirm: 'Are you sure?' } %>
+        <% end %>
```
Comment (nwalberts) [inline app/views/restaurants/index.html.erb:10]: technically part of another feature T_T

File: spec/features/restaurants/index_page_spec.rb [hunk 6]
```diff
@@ -1,18 +1,36 @@
 require 'rails_helper'
 
 feature "Going to Restaurant Index page" do
-    let!(:restaurant1) { FactoryBot.create(:restaurant) }
-    let!(:restaurant2) { FactoryBot.create(:restaurant) }
+  let!(:restaurant1) { FactoryBot.create(:restaurant) }
+  # let!(:restaurant2) { FactoryBot.create(:restaurant) }
```
Comment (nwalberts) [inline spec/features/restaurants/index_page_spec.rb]: if you arent using this review that is commented out, just delete it

File: spec/javascript/editReviewFormSpec.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        wrapper.find('.edit-submit').first().simulate('submit')
+        setTimeout(() => {
+          expect(wrapper.find('.edit-review')).toBePresent()
+          expect(wrapper.find('.review-tile').nodes.length).toEqual(restaurant.restaurant.reviews.length - 1)
+          done()
+        })
+      })
+    }, 0)
+  })
+})
```
Comment (nwalberts) [inline spec/javascript/editReviewFormSpec.js:122]: DAMN. Nice job getting this working.

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: app/javascript/containers/ReviewContainer.js (hunk 3 above)
Comment (gilhsu) [reply app/javascript/containers/ReviewContainer.js:74]: We made this into a function to be DRYer.

**Pull request merged**